src/Sparrow.Server/LowMemory/CheckPageFileOnHdd.cs

```diff
@@ -391,17 +394,31 @@ public static string PosixIsSwappingOnHddInsteadOfSsd()
                 if (swaps.Length == 0) // on error return as if no swap problem
                     return null;
 
+                const string sysBlockDirectoryPath = "/sys/block/";
+
+                var blocks = Directory
+                    .GetDirectories(sysBlockDirectoryPath)
+                    .Select(x => x.Substring(sysBlockDirectoryPath.Length))
+                    .Where(x => x.StartsWith("loop") == false)
+                    .ToList();
+
+                if (blocks.Count == 0)
+                    return null;
+
                 string foundRotationalDiskDrive = null;
                 for (int i = 0; i < swaps.Length; i++)
                 {
                     if (swaps[i].IsDeviceSwapFile)
                         continue; // we do not check swap file, only partitions
 
-                    // remove numbers at end of string (i.e.: /dev/sda5 ==> sda)
-                    var disk = _regExRemoveNumbers.Replace(swaps[i].DeviceName, "").Replace("/dev/", "");
+                    if (TryFindDisk(swaps[i].DeviceName, out var disk) == false)
+                        return null;
+
                     var filename = $"/sys/block/{disk}/queue/rotational";
-                    var isHdd = KernelVirtualFileSystemUtils.ReadNumberFromFile(filename);
+                    if (File.Exists(filename) == false)
+                        return null;
 
+                    var isHdd = KernelVirtualFileSystemUtils.ReadNumberFromFile(filename);
                     if (isHdd == -1)
                         return null;
```

Comment:
How this can return -1? It should only be 0 or 1? On error this method returns long.MaxValue